backend-lib/src/main/rust/lib.rs

```diff
@@ -443,8 +443,8 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_createAcc
     let res = catch_unwind(&mut env, |env| {
         let network = parse_network(network_id as u32)?;
         let mut db_data = wallet_db(env, network, db_data)?;
-        let seed = SecretVec::new(env.convert_byte_array(seed).unwrap());
-        let treestate = TreeState::decode(&env.convert_byte_array(treestate).unwrap()[..])
+        let seed = parse_secret(env, &seed)?;
+        let treestate = TreeState::decode(utils::java_bytes_to_rust(env, &treestate)?.as_slice())
             .map_err(|e| anyhow!("Invalid TreeState: {}", e))?;
```

Comment:
There are three identical uses of this pattern. Add a `treestate_from_jni` helper.